ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+
+        # write all data into the file
+        # write the file
+        data = self.location.ep_style_location_string + '\n\n'
+        for d_day in self.design_days:
+            data = data + d_day.ep_style_string + '\n\n'
+        write_to_file(full_path, data, True)
+
+    def filter_by_keyword(self, keyword):
+        """Return a list of ddys that have a certain keyword in their name.
+
+        This is useful for selecting out design days from a ddy file that are
+        for a specific type of condition (for example, .4% cooling design days)
+        """
+        filtered_days = []
+        for des_day in self.design_days:
+            if keyword in des_day.name:
+                filtered_days.append(des_day)
+        return filtered_days
+
+    @property
+    def file_path(self):
+        """Get or set path to ddy file."""
+        return self._file_path
+
+    @property
+    def folder(self):
+        """Get ddy file folder."""
+        return self._folder
+
+    @property
+    def file_name(self):
+        """Get ddy file name."""
+        return self._file_name
+
+    @file_path.setter
+    def file_path(self, ddy_file_path):
+        self._file_path = os.path.normpath(ddy_file_path)
+
+        if not ddy_file_path.lower().endswith('.ddy'):
+            self._file_path = self._file_path + '.ddy'
+
+        self._folder, self._file_name = os.path.split(self.file_path)
+
+    @property
+    def location(self):
+        """Get or set the location."""
+        return self._location
+
+    @location.setter
+    def location(self, data):
+        assert hasattr(data, 'isLocation'), 'Expected' \
+            ' Location type. Got {}'.format(str(data))
+        self._location = data
+
+    @property
+    def design_days(self):
+        """Get or set the design_days."""
+        return self._design_days
+
+    @design_days.setter
+    def design_days(self, data):
+        assert isinstance(data, list), 'Expected' \
+            ' a list of design days. Got {}'.format(str(data))
+        for item in data:
+            assert hasattr(item, 'isDesignDay'), 'Expected' \
+                ' DesignDay type. Got {}'.format(str(item))
+        self._design_days = data
+        for dd in self._design_days:
+            dd.location = self._location
+
+    @property
+    def isDdy(self):
+        """Return True."""
+        return True
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __repr__(self):
+        """Ddy object representation."""
+        return "Ddy File - {} [# days: {}]".format(
+            self.location.city, str(len(self._design_days)))
+
+
+# TODO: add a property for hourly horizontal infrared radiation.
+class DesignDay(object):
+    """An object representing design day conditions.
+
+    properties:
+        name: A text string to set the name of the design day
+        day_type: Choose from 'SummerDesignDay', 'WinterDesignDay' or other
```

Comment:
What are other EnergyPlus days and how can one get it? Now that I see this I think it is good to have it as a property or something like that.

Reply:
Yes, I've put it as a property and referenced it in the doc string.